processor/metricstransformprocessor/metrics_transform_processor_helper.go

```diff
@@ -0,0 +1,372 @@
+// Copyright 2020 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metricstransformprocessor
+
+import (
+	"fmt"
+	"math"
+	"math/rand"
+
+	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
+	"github.com/golang/protobuf/ptypes/timestamp"
+)
+
+// timeseriesGroupByLabelValues is a data structure for grouping timeseries that will be aggregated
+type timeseriesGroupByLabelValues struct {
+	// key is a composite of the label values as a single string
+	// keyToTimeseriesMap groups timeseries by the label values
+	keyToTimeseriesMap map[string][]*metricspb.TimeSeries
+	// keyToLabelValuesMap maps the keys to the actual label values objects
+	keyToLabelValuesMap map[string][]*metricspb.LabelValue
+}
+
+// update updates the metric content based on operations indicated in transform.
+func (mtp *metricsTransformProcessor) update(metric *metricspb.Metric, transform Transform) {
+	// metric name update
+	if transform.NewName != "" {
+		metric.MetricDescriptor.Name = transform.NewName
+	}
+
+	for _, op := range transform.Operations {
+		switch op.Action {
+		case UpdateLabel:
+			mtp.updateLabelOp(metric, op)
+		case AggregateLabels, AggregateLabelValues:
+			mtp.aggregateOp(metric, op)
+		}
```

Comment:
Nice, separated. However each of the operation files only has one function because update label operation is an easy operation, and the two aggregation operations have everything overlapped.

Reply:
Much better imo! 😄